web/assets/js/components/gameboard.ts

```diff
@@ -1,232 +1,247 @@
-        this.DOM.moves = [];
-      }
-
-      // get allowed position on board DOM then DOM
-      this.selected = currentCell;
-      const moves = this.chess.moves({ square: currentCell, verbose: true });
-      this.moves = [...new Set(moves.map((pos: any) => pos.to))]; //remove duplicate of "to" (promotion ones)
-      this.promotion = [...new Set(moves.filter((move: any) => move.promotion).map((move: any) => move.to))];
-
-      this.moves.forEach((cell: Element) => {
-        this.DOM.moves.push(this.DOM.el.querySelector(`[data-square="${cell}"]`));
-      });
-      // highlight allowed position
-      if (this.DOM.moves.length > 0) gsap.to(this.DOM.moves, { "--disp-opacity": 1 });
+
+    showScoreBoard(winner: Colors) {
+        gsap.set(this.DOM.el, {willChange: "transform"});
+        gsap.to(this.DOM.el, {
+            rotate: winner === "w" ? "-20deg" : "20deg",
+            x: "60%",
+            y: winner === "w" ? "0%" : "25%",
+            scale: 0.8
+        });
+    }
+
+    getMoveNumber() {
+        return this.chess.moveNumber();
+    }
+
+    async engine(init = false, gameover?: GameoverType) {
+        //for test purpose -> to et in if statment bellow
+        // setTimeout(() => {
+        //   this.call("finishGame", "gameover", "gameplayers", "me");
+        //   this.call("stopTimer", [true], "gameplayers", "me");
+        //   this.call("stopTimer", [true], "gameplayers", "rival");
+        //   this.call("disappear", "", "gamecontrols");
+        //   this.showScoreBoard("w");
+        // }, 10000);
+
+        const actions: Actions = await Actions.getInstance()
+
+        //TODO: win cause rival resign
+        //TODO: draw screen in players
+        if (this.chess.isGameOver() || gameover) {
+            console.log("GAME OVER!");
+            let status: GameoverType = "checkmate";
+
+            if (gameover === "timeout") {
+                status = "timeout";
+                const valid = await actions.isGameOver("", "timeout"); // TODO @Alexis add game ID
+                if (valid) this.call("finishGame", ["winner", status], "gameplayers", this.color === this.chess.turn() ? "rival" : "me");
+            }
+
+            if (this.chess.isCheckmate()) {
+                status = "checkmate";
+                const valid = await actions.isGameOver("", "timeout"); // TODO @Alexis add game ID, change type (to checkmate)?
+                if (valid) this.call("finishGame", ["winner", status], "gameplayers", this.color === this.chess.turn() ? "me" : "rival");
+            }
+
+            if (this.chess.isDraw() || gameover === "draw") {
+                status = this.chess.isStalemate() ? "stalemate" : this.chess.isThreefoldRepetition() ? "threefoldRepetition" : this.chess.isInsufficientMaterial() ? "insufficientMaterial" : "draw";
+                const valid = await actions.isGameOver("", "timeout"); // TODO @Alexis add game ID, change type (to stalemate)?
```

Comment:
Regarding the change type, the status can be "stalemate", "ThreefoldRepetition", "insufficientMaterial" or "draw" (all are kind of draw). Should we provide "draw" anyway?

Reply:
I'm guessing so, or maybe we just add an all-encompassing game ended check (check all types that conclude the game)